src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java

```diff
@@ -474,6 +477,75 @@ public StudyDetailsDto getStudyDetailsByInstance(final Integer instanceId) {
+		try {
+			final List<StudyInstanceDto> studyInstanceDtos = this.daoFactory.getDmsProjectDAO()
+				.getStudyInstances(studySearchFilter, pageable);
+			final Map<Integer, List<ObservationLevel>> trialIdToObservationLevelMap = new HashMap<>();
+			if(studyInstanceDtos != null) {
+				for(final StudyInstanceDto studyInstanceDto: studyInstanceDtos) {
+					final Integer trialDbId = Integer.valueOf(studyInstanceDto.getTrialDbId());
+					final Integer studyDbId = Integer.valueOf(studyInstanceDto.getStudyDbId());
+					final DmsProject environmentDataset =
+						this.daoFactory.getDmsProjectDAO()
+							.getDatasetsByTypeForStudy(trialDbId, DatasetTypeEnum.SUMMARY_DATA.getId()).get(0);
+					final List<MeasurementVariable> environmentConditions = this.daoFactory.getDmsProjectDAO()
+						.getObservationSetVariables(
+							environmentDataset.getProjectId(),
+							Lists.newArrayList(VariableType.ENVIRONMENT_CONDITION.getId()));
+					final List<MeasurementVariable> environmentParameterVariables = new ArrayList<>();
+					List<Integer> variableIds = this.getVariableIds(environmentConditions);
+					if (!variableIds.isEmpty()) {
+						environmentParameterVariables.addAll(
+							this.studyDataManager
+								.getEnvironmentConditionVariablesByGeoLocationIdAndVariableIds(studyDbId, variableIds));
+					}
+					final List<MeasurementVariable> environmentDetails = this.daoFactory.getDmsProjectDAO()
+						.getObservationSetVariables(
+							environmentDataset.getProjectId(),
+							Lists.newArrayList(VariableType.ENVIRONMENT_DETAIL.getId()));
+					variableIds = this.getVariableIds(environmentDetails);
+					if (!variableIds.isEmpty()) {
+						environmentParameterVariables.addAll(
+							this.studyDataManager
+								.getEnvironmentDetailVariablesByGeoLocationIdAndVariableIds(studyDbId, variableIds));
+					}
+
+					final List<MeasurementVariable> environmentVariables = new ArrayList<>(environmentConditions);
+					environmentVariables.addAll(environmentDetails);
+					environmentParameterVariables.addAll(this.createGeolocationVariables(environmentVariables, studyDbId));
+
+					final List<EnvironmentParameter> environmentParameters = environmentParameterVariables.stream()
+						.map(variable -> new EnvironmentParameter(variable)).collect(Collectors.toList());
+					studyInstanceDto.setEnvironmentParameters(environmentParameters);
+
+					variableIds = this.getVariableIds(environmentVariables);
+					studyInstanceDto.getAdditionalInfo()
+						.putAll(this.studyDataManager.getGeolocationPropsAndValuesByGeolocation(studyDbId, variableIds));
+					studyInstanceDto.getAdditionalInfo()
+						.putAll(this.studyDataManager.getProjectPropsAndValuesByStudy(trialDbId, variableIds));
+
+					if(trialIdToObservationLevelMap.get(studyInstanceDto.getTrialDbId()) == null) {
+						final List<DatasetDTO> datasetDTOs = this.daoFactory.getDmsProjectDAO().getDatasets(trialDbId);
```

Comment:
I think this MW call for getDatasets is unnecessary. You can use the `environmentDataset` you retrieved before.